src/SharpCompress/Algorithms/Adler32.cs

```diff
@@ -0,0 +1,420 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+#if !NETSTANDARD2_0 && !NETSTANDARD2_1 && !NETFRAMEWORK
+#define SUPPORTS_RUNTIME_INTRINSICS
```

Comment:
This is my guess of how the #defines should work based on the previous code in Alder32.cs and this in https://github.com/SixLabors/SharedInfrastructure/blob/main/msbuild/props/SixLabors.Global.props

```
    <When Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)','netcoreapp3.1'))">
      <!--NETCORE 3.1. NET5.0, and future versions will fallback to this as the closest target.-->
      <PropertyGroup>
...
        <DefineConstants>$(DefineConstants);SUPPORTS_RUNTIME_INTRINSICS</DefineConstants>
        <DefineConstants>$(DefineConstants);SUPPORTS_HOTPATH</DefineConstants>
...
      </PropertyGroup>
    </When>
```